### PR TITLE
Add `middle-ellipsis` component and use it in default list item MAB-13670

### PR DESCRIPTION
**Pull request by zigomir**
https://celtra.atlassian.net/browse/MAB-13670

I'm aware we're using `middleEllipsis` filter elsewhere still, but I just want to open this as a POC that works well for Campaign Explorer.

I think current solution with `calculateWidths` is too hacky and doesn't work well in many situations. Lets try to converge on a component that works for all use cases. Please let me know your feedback.

### CODE REVIEW

File: src/stateless/MiddleEllipsis.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                if (percent < 1) {
+                    return middleEllipsis(this.text, breakAt)
+                }
+            }
+
+            return this.text
+        },
+    },
+    mounted () {
+        this.width = parseInt(window.getComputedStyle(this.$refs['middle-ellipsis']).width, 10)
```
Comment (zan-kusterle) [inline src/stateless/MiddleEllipsis.vue]: Maybe you should also update this on resize event and make it reactive.

**Comment by zigomir** [comment]
@zan-kusterle cool, thanks! I see that https://github.com/celtra/stateless-components/pull/3 was closed. Should I point this PR on to another branch now?

**Comment by zan-kusterle** [comment]
Wait for `report-builder-fresh` to be merged to `master` and change base to `master` then.

File: src/stateless/MiddleEllipsis.vue (hunk 1 above)
Comment (zigomir) [reply src/stateless/MiddleEllipsis.vue]: Maybe, but I'd only add it when it's needed, not before.

**Comment by zigomir** [comment]
@zan-kusterle I updated snapshots so Circle is passing. Please let me know if this would play well with report builder.

**Comment by zan-kusterle** [comment]
This does not solve the same issue we want to solve with DefaultListItem. With DefaultListItem we don't want to add ellipsis to the label and metadata at the same time. This solution adds middle ellipsis to the metadata instead of removing a few extra characters from the label (last item):
screen shot 2018-09-10 at 11 25 52

**Comment by zigomir** [comment]
@zan-kusterle all right. I isolated changes to `MiddleEllipsisListItem` so anything that's not using `ChipWithMultiselect` won't be affected (DefaultListItem is staying the same). I hope a bit of duplication isn't a deal breaker (mostly size styles).

**Pull request merged**